controller/api/destination/profile_translator.go

```diff
@@ -43,10 +46,26 @@ var profileUpdatesQueueOverflowCounter = promauto.NewCounterVec(
 	},
 )
 
-func newProfileTranslator(stream pb.Destination_GetProfileServer, log *logging.Entry, fqn string, port uint32, endStream chan struct{}) *profileTranslator {
+func newProfileTranslator(serviceID watcher.ServiceID, stream pb.Destination_GetProfileServer, log *logging.Entry, fqn string, port uint32, endStream chan struct{}) *profileTranslator {
+	var parentRef *meta.Metadata
+	if serviceID.Namespace != "" {
```

Comment:
Is there any codepath where this is reasonably expected to be empty? I believe places that these IDs are constructed are `parseK8sServiceName` which extracts the namespace from a fqdn with a validated number of segments and `getSvcID` which gets the namespace from a Service resource object.  In both cases the namespace should be non-empty unless something pathological is happening.

Reply:
when looking up a non-service IP

Reply:
I believe that will use the endpointProfileTranslator instead of the profileTranslator